services/security/authentication/casigned/casigned.go

```diff
@@ -37,8 +46,49 @@ func (c *client) Login(ctx context.Context, group string, identity *security.Ide
 	return &response.Token, nil
 }
 
+func RenewRoutine(ctx context.Context, group, server string) {
+	renewalAttempt := 0
+	// Waiting for a few seconds to avoid spamming short-lived sdk user
+	time.Sleep(time.Second * 5)
+	var err error
+	_, span := trace.NewSpan(ctx, "RenewRoutine")
+	defer span.End(err)
+	for {
+		wssdConfig := auth.WssdConfig{}
+		err := marshal.FromJSONFile(auth.GetWssdConfigLocation(), &wssdConfig)
+		if err != nil {
+			span.Log("Failed to open config file in location %s: %v\n", auth.GetWssdConfigLocation(), err)
+			return
+		}
+
+		sleepTime, renewalBackoff, expiry, err := renewTime(wssdConfig.ClientCertificate)
+		if err != nil {
+			span.Log("Failed while calculating certificate renew time %v \n", err)
+			return
+		}
+		span.Log("Waiting for %v to renew cert\n", sleepTime)
+		time.Sleep(sleepTime)
+		span.Log("Attempting to renew certificate\n")
+		err = auth.RenewCertificates(server, auth.GetWssdConfigLocation())
+		if err != nil {
+			if errors.IsExpired(err) {
+				span.Log("Certificate Expired %v", err)
```

Comment:
Do we have context here to perform an auto login?

Reply:
We do have but the token we have could be expired most likely as the token validity is likely less than the cert validity

Reply:
If auto rotation was done, the possibility of token expiry is even reduced and trying to login back should get the cases of panic even lesser

Reply:
lets handle that is next PR